src/pages/progress_report/SubmitProgressReport.tsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import {AxiosError} from 'axios';
+import {ProgressReport} from '../../libs/ajax/ProgressReport';
+import {Notifications} from '../../libs/utils';
+import {ConsentError} from '../../types/responseTypes';
+
+
+interface SubmitProgressReportProps {
+  readonly progressReport: object;
+  readonly parentReferenceId: string;
+  readonly onSuccess: (result: unknown) => void;
+  readonly onCancel: (result: unknown) => void;
+}
+
+export default function SubmitProgressReport(props: SubmitProgressReportProps) {
+  const {progressReport, parentReferenceId, onSuccess, onCancel} = props;
+
+  const submit = async () => {
+    try {
+      const submittedPR = await ProgressReport.submitProgressReport(progressReport, parentReferenceId);
+      onSuccess(submittedPR);
+    } catch (error: unknown) {
+      handleError('Error: Unable to submit progress report: ', error);
+    }
+  }
+
+  const cancel = async () => {
+    try {
+      onCancel(progressReport);
+    } catch (error: unknown) {
+      handleError('Error: Unable to cancel progress report submission: ', error);
+    }
+  }
+
+  const handleError = (message: string, error: unknown): void => {
+    const axiosError = error as AxiosError;
+    const consentError = axiosError?.response?.data as ConsentError;
+    const serverError = consentError.message ?? 'Unknown error';
+    Notifications.showError({text: message + serverError});
+  }
```

Comment:
I'm especially interested in the error handling. I found this to be non-trivial to work through. I wonder if there is a better pattern to follow so that we don't swallow server-provided errors.